OpenRA.Mods.RA/Widgets/Logic/OrderButtonsChromeLogic.cs

```diff
@@ -101,7 +101,7 @@ public OrderButtonsChromeLogic(Widget widget, World world)
 			if (diplomacy != null)
 			{
 				diplomacy.Visible = world.Players.Any(a => a != world.LocalPlayer && !a.NonCombatant);
-				diplomacy.IsDisabled = () => disableSystemButtons;
+				diplomacy.IsDisabled = () => world.Map.Type == "Mission" || world.Map.Type == "Campaign" || disableSystemButtons;
```

Comment:
We now have a proper map class enum which should be used instead of these magic type values.